### ISSUE TITLE
how to search sub projects?

### ISSUE THREAD

**Issue by rileyrg** [bbatsov/projectile#1575]
my .emacs.d is a projectile directory. I use straight.el for package management.
.emacs.d/.gitgnore ignores straight/repos as they are all individual git projects.

I added .git to my projectile root files

> projectile-project-root-files is a variable defined in `projectile.el'.
> Its value is 
> ("pubspec.yaml" "info.rkt" "Cargo.toml" "stack.yaml" "DESCRIPTION" "Eldev" "Cask" "shard.yml" "Gemfile" ".bloop" "deps.edn" "build.boot" "project.clj" "build.sbt" "application.properties" "gradlew" "build.gradle" "pom.xml" "poetry.lock" "Pipfile" "tox.ini"
> "setup.py" "requirements.txt" "manage.py" "angular.json" "package.json" "gulpfile.js" "Gruntfile.js" "mix.exs" "rebar.config" "composer.json" "CMakeLists.txt" "Makefile" "WORKSPACE" "default.nix" "meson.build" "SConstruct" "GTAGS" "TAGS" "configure.ac"
> "configure.in" "cscope.out" ".git")
> 
> Original value was 
> 
> ("GTAGS" "TAGS" "configure.ac" "configure.in" "cscope.out")
> 

When I open emacs and open .emacs.d/straight/repos/my-proj/myfile.el  the projectile root is still .emacs.d. Why? my-proj has a .git directory. I also tried adding .gitignore and that didn't work either. HOWEVER when I added a .projectile it did work (when opening myfile.el as the first file after starting emacs)  and the project root was emacs.d/straight/repos/my-proj.  

When the current project root is ~/.emacs.d despite me editing my-repo/myfile.el I tried to use projectile-switch-project (helm completion) and that doesn't work - despite offering .emacs.d/straight/repos/my-repo as a project to switch to.

Why do I want to do this? I want to be able to us helm-projectile-ag to search sub projects. Currently the top level .gitignore excludes them since they are not part of the top level git repo. Yes, I can prefix helm-projectile-ag and enter "-u" to ignore .gitignore but this then searches caches and logs and so forth - very messy.

Bottom line : I want to use a fast grep to include sub projects (git repos) that may be excluded in .gitigore.

Hints and tips much appreciated.

**Comment by rileyrg** [comment]
I cant cope with the -U option
